### ISSUE TITLE
[ChunkStore] chunk_range not closed on the right with timestamp indexes

### ISSUE THREAD

**Issue by alberto-exabel** [manahl/arctic#709]
#### Arctic Version

```
1.74.0
```

#### Arctic Store

```
ChunkStore
```

#### Platform and version

```
Ubuntu 18.04.1 LTS
```

#### Description of problem and/or code sample that reproduces the issue

If the date index of the data in _ChunkStore_ has **timestamps**, when reading it with the `chunk_range` parameter and a time range of dates, the time range is effectively interpreted as being closed on the start date and **open** on the end date.
If the date index of the data has **dates** instead, when reading it with the `chunk_ranges` parameter and a time range of dates, the time range is effectively interpreted as being closed on the start date and **closed** on the end date.
This also happens when using the parameter `chunk_range` in the `update` method.

Example code:
```
from arctic import Arctic, CHUNK_STORE
from arctic.date import DateRange
import pandas as pd

Arctic('127.0.0.1').initialize_library('test.daily', lib_type=CHUNK_STORE)
df = pd.DataFrame(
    {'A': [1, 2, 3]},
    index=pd.DatetimeIndex(
        pd.date_range('2019-01-01', periods=3, freq='D'),
        name='date'
        )
    )
Arctic('127.0.0.1')['test.daily'].write('test', df, chunk_size='D')

Arctic('127.0.0.1').initialize_library('test.intraday', lib_type=CHUNK_STORE)
df = pd.DataFrame(
    {'A': [1, 2, 3]},
    index=pd.DatetimeIndex(
        pd.date_range('2019-01-01 08:00:00', periods=3, freq='H'),
        name='date'
        )
    )
Arctic('127.0.0.1')['test.intraday'].write('test', df, chunk_size='D')

Arctic('127.0.0.1')['test.daily'].get_info('test')
Arctic('127.0.0.1')['test.intraday'].get_info('test')

list(Arctic('127.0.0.1')['test.daily'].get_chunk_ranges('test'))
list(Arctic('127.0.0.1')['test.intraday'].get_chunk_ranges('test'))

Arctic('127.0.0.1')['test.daily'].read(
    'test',
    chunk_range=DateRange('2019-01-01', '2019-01-01')
    )
Arctic('127.0.0.1')['test.intraday'].read(
    'test',
    chunk_range=DateRange('2019-01-01', '2019-01-01')
    )
Arctic('127.0.0.1')['test.intraday'].read(
    'test',
    chunk_range=DateRange('2019-01-01', '2019-01-02')
    )

df = pd.DataFrame(
    {'A': [1]},
    index=pd.DatetimeIndex(
        pd.date_range('2019-01-01', periods=1, freq='D'),
        name='date'
        )
    )
Arctic('127.0.0.1')['test.daily'].update(
    'test',
    df,
    chunk_size='D',
    upsert=True,
    chunk_range=DateRange('2019-01-01', '2019-01-01')
    )

df = pd.DataFrame(
    {'A': [1, 2, 3]},
    index=pd.DatetimeIndex(
        pd.date_range('2019-01-01 08:00:00', periods=3, freq='H'),
        name='date'
        )
    )
Arctic('127.0.0.1')['test.intraday'].update(
    'test',
    df,
    chunk_size='D',
    upsert=True,
    chunk_range=DateRange('2019-01-01', '2019-01-01')
    )
```

**Comment by alberto-exabel** [comment]
My point here is that it would be convenient to have date ranges read as closed on the right, so that we don't need to write ```DateRange('2019-01-01 00:00:00', '2019-01-02 23:59:59')``` but instead we can just write ```DateRange('2019-01-01', '2019-01-02')```

**Comment by bmoscon** [comment]
The issue here is that if you dont give it a timestamp, pandas assumes its 00:00:00. If you want to read just jan 1st, you can use DateRange('2019-01-01', '2019-01-02'), if you want jan2nd too, you can do DateRange('2019-01-01', '2019-01-03'), because pandas will effectively make that DateRange('2019-01-01 00:00:00' , '2019-01-03 00:00:00)

**Comment by bmoscon** [comment]
also, @jamesblackburn changed the title to ChunkStore, but I think anything using the DateRange will fail like there are timestamps in the datetime index (I could be wrong)

**Comment by alberto-exabel** [comment]
Ok, if the user is expected to pass timestamps in a date range for a timestamp index or at least to be aware that when passing dates instead of timestamps in the date range, the dates will be used as timestamps with 00:00:00, then there is no need to handle in the code the latter case. It sounds like a reasonable expectation. I will close this issue and start to use the date range parameter properly. My intention was to automate the usage of dates in the a range for timestamp indexes, so that users don't fall into that trap as I did the first time I used it. Thanks for looking at it.
